src/test/java/io/jenkins/plugins/opentelemetry/JenkinsOtelPluginIntegrationTest.java

```diff
@@ -167,15 +170,46 @@ public void testTraceEnvironmentVariablesInjectedInShellSteps() throws Exception
                     "'''\n" +
                     "    }\n" +
                     "}";
+            final Node node = jenkinsRule.createOnlineSlave();
+
             WorkflowJob pipeline = jenkinsRule.createProject(WorkflowJob.class, "test-trace-environment-variables-injected-in-shell-steps-" + jobNameSuffix.incrementAndGet());
             pipeline.setDefinition(new CpsFlowDefinition(pipelineScript, true));
             WorkflowRun build = jenkinsRule.assertBuildStatus(Result.SUCCESS, pipeline.scheduleBuild2(0));
 
             final Tree<SpanDataWrapper> spans = getGeneratedSpans();
-            MatcherAssert.assertThat(spans.cardinality(), CoreMatchers.is(6L));
+            MatcherAssert.assertThat(spans.cardinality(), CoreMatchers.is(8L));
         }
     }
 
+    @Test
+    public void testPipelineWithNodeSteps() throws Exception {
+        String pipelineScript = "pipeline {\n" +
+                "  agent none\n" +
+                "  stages {\n" +
+                "    stage('foo') {\n" +
+                "      steps {\n" +
+                "        node('linux') { \n" +
+                "          echo 'hello world' \n" +
+                "        }\n" +
+                "      }\n" +
+                "    }\n" +
+                "  }\n" +
+                "}";
+
+        final Node node = jenkinsRule.createOnlineSlave();
+        node.setLabelString("linux");
+
+        final String jobName = "test-simple-pipeline-" + jobNameSuffix.incrementAndGet();
+        WorkflowJob pipeline = jenkinsRule.createProject(WorkflowJob.class, jobName);
+        pipeline.setDefinition(new CpsFlowDefinition(pipelineScript, true));
+        WorkflowRun build = jenkinsRule.assertBuildStatus(Result.SUCCESS, pipeline.scheduleBuild2(0));
+
+        Tree<SpanDataWrapper> spans = getGeneratedSpans();
+        checkChainOfSpans(spans, "Phase: Start", jobName);
+        checkChainOfSpans(spans, "Node: Ready", "Node: Allocate (linux)", "Stage: foo", "Phase: Run");
+        checkChainOfSpans(spans, "Phase: Finalise", jobName);
+        MatcherAssert.assertThat(spans.cardinality(), CoreMatchers.is(7L));
```

Comment:
Can we check that we create a span attribute that reports the node label `linux`?

Reply:
See https://github.com/jenkinsci/opentelemetry-plugin/pull/22/commits/ae0e6ab17e64d96dc34d0a09da52f1027072308e